hw/xbox/nv2a/nv2a_stubs.c

```diff
@@ -19,100 +19,29 @@
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PRMA, addr, val);
-}
-
-uint64_t pcounter_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PCOUNTER, addr, 0);
-    return 0;
-}
-void pcounter_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PCOUNTER, addr, val);
-}
-
-uint64_t pvpe_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PVPE, addr, 0);
-    return 0;
-}
-void pvpe_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PVPE, addr, val);
-}
-
-uint64_t ptv_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PTV, addr, 0);
-    return 0;
-}
-void ptv_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PTV, addr, val);
-}
-
-uint64_t prmfb_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PRMFB, addr, 0);
-    return 0;
-}
-void prmfb_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PRMFB, addr, val);
-}
-
-uint64_t prmdio_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PRMDIO, addr, 0);
-    return 0;
-}
-void prmdio_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PRMDIO, addr, val);
-}
-
-uint64_t pstraps_read(void *opaque,
-                                  hwaddr addr, unsigned int size)
-{
-    reg_log_read(NV_PSTRAPS, addr, 0);
-    return 0;
-}
-void pstraps_write(void *opaque, hwaddr addr,
-                               uint64_t val, unsigned int size)
-{
-    reg_log_write(NV_PSTRAPS, addr, val);
-}
-
-/* PRAMIN - RAMIN access */
-/*
-uint64_t pramin_read(void *opaque,
-                                 hwaddr addr, unsigned int size)
-{
-    NV2A_DPRINTF("nv2a PRAMIN: read [0x%" HWADDR_PRIx "] -> 0x%" HWADDR_PRIx "\n", addr, r);
-    return 0;
-}
-void pramin_write(void *opaque, hwaddr addr,
-                              uint64_t val, unsigned int size)
-{
-    NV2A_DPRINTF("nv2a PRAMIN: [0x%" HWADDR_PRIx "] = 0x%02llx\n", addr, val);
-}*/
+#define DEFINE_STUB(name, region_id) \
+    uint64_t name ## _read(void *opaque, \
+                           hwaddr addr, \
+                           unsigned int size) \
+    { \
+        reg_log_read(region_id, addr, 0); \
+        return 0; \
+    } \
+    void name ## _write(void *opaque, \
+                        hwaddr addr, \
+                        uint64_t val, \
+                        unsigned int size) \
+    { \
+        reg_log_write(region_id, addr, val); \
+    } \
```

Comment:
nit: Extra backslash slightly dangerous in my opinion

Reply:
This was not done. I'd appreciate if it was done, as at least one other multi-line macro in this change doesn't seem to have it either.